src/lib.rs

```diff
@@ -226,30 +185,72 @@ impl Order {
     }
 }
 
+/// An stream-like interface that yields an [`Order`]'s authoritations
+///
+/// Call [`next()`] to get the next authorization in the order. If the order state
+/// does not yet contain the state of the authorization, it will be fetched from the
+/// server. If the authorization is pending, you can set up a challenge response.
+///
+/// [`next()`]: Authorizations::next()
+pub struct Authorizations<'a> {
+    iter: slice::IterMut<'a, Authorization>,
+    nonce: &'a mut Option<String>,
+    account: &'a AccountInner,
+}
+
+impl Authorizations<'_> {
+    /// Yield the next [`AuthorizationHandle`], fetching its state if we don't have it yet.
+    pub async fn next(&mut self) -> Option<Result<AuthorizationHandle<'_>, Error>> {
+        let authz = self.iter.next()?;
+        if authz.state.is_none() {
+            match self.account.get(self.nonce, &authz.url).await {
+                Ok(state) => authz.state = Some(state),
+                Err(e) => return Some(Err(e)),
+            }
+        }
+
+        Some(Ok(AuthorizationHandle {
+            // This is a little ugly, but I don't think we can do better in 2021 edition.
+            // Seems like there are other better options with the scope changes in 2024,
+            // which we can't use yet due to MSRV limitations.
+            state: authz.state.as_mut().unwrap(),
+            url: &authz.url,
+            nonce: self.nonce,
+            account: self.account,
+        }))
+    }
+}
+
 /// An ACME authorization as described in RFC 8555 (section 7.1.4)
 ///
 /// Authorizations are retrieved from an associated [`Order`] by calling
-/// [`Order::authorizations()`].
+/// [`Order::authorizations()`]. This type dereferences to the underlying
+/// [`AuthorizationState`] for easy access to the authorization's state.
+///
+/// For each authorization, you'll need to:
+///
+/// * Select which [`ChallengeType`] you want to complete
+/// * Call [`AuthorizationHandle::challenge()`] to get a [`ChallengeHandle`]
```

Comment:
maybe missing another bullet point saying something like "* Use the `ChallengeHandle` to complete the authorization's challenge" ?